packages/console/src/pages/ConnectorDetails/components/ConnectorContent.tsx

```diff
@@ -135,9 +135,14 @@ const ConnectorContent = ({ isDeleted, connectorData, onConnectorUpdated }: Prop
         {connectorData.type !== ConnectorType.Social && (
           <FormCard title="connector_details.test_connection">
             <SenderTester
-              connectorId={connectorData.id}
+              connectorFactoryId={connectorData.connectorId}
               connectorType={connectorData.type}
-              config={watch('config')}
+              formConfig={
+                connectorData.formItems
+                  ? parseFormConfig(watch(), connectorData.formItems)
+                  : undefined
+              }
```

Comment:
conditional(connectorData.formItems && parseFormConfig(watch(), connectorData.formItems)) ?

Reply:
nit: could pass in a onSubmit  useCallback hook, instead of pass in the config, and update on every cycle.

Reply:
Updated using `conditional()`.
Added LOG-5800 to track the refactor.